// rive-react-native/lib/module/types.js
export let Fit;

(function (Fit) {
  Fit["Cover"] = "cover";
  Fit["Contain"] = "contain";
  Fit["Fill"] = "fill";
  Fit["FitWidth"] = "fitWidth";
  Fit["FitHeight"] = "fitHeight";
  Fit["None"] = "none";
  Fit["ScaleDown"] = "scaleDown";
})(Fit || (Fit = {}));

export let Alignment;

(function (Alignment) {
  Alignment["TopLeft"] = "topLeft";
  Alignment["TopCenter"] = "topCenter";
  Alignment["TopRight"] = "topRight";
  Alignment["CenterLeft"] = "centerLeft";
  Alignment["Center"] = "center";
  Alignment["CenterRight"] = "centerRight";
  Alignment["BottomLeft"] = "bottomLeft";
  Alignment["BottomCenter"] = "bottomCenter";
  Alignment["BottomRight"] = "bottomRight";
})(Alignment || (Alignment = {}));

export let LoopMode;

(function (LoopMode) {
  LoopMode["OneShot"] = "oneShot";
  LoopMode["Loop"] = "loop";
  LoopMode["PingPong"] = "pingPong";
  LoopMode["Auto"] = "auto";
})(LoopMode || (LoopMode = {}));

export let Direction;

(function (Direction) {
  Direction["Backwards"] = "backwards";
  Direction["Auto"] = "auto";
  Direction["Forwards"] = "forwards";
})(Direction || (Direction = {}));

export let RNRiveErrorType;

(function (RNRiveErrorType) {
  RNRiveErrorType["FileNotFound"] = "FileNotFound";
  RNRiveErrorType["UnsupportedRuntimeVersion"] = "UnsupportedRuntimeVersion";
  RNRiveErrorType["IncorrectRiveFileUrl"] = "IncorrectRiveFileUrl";
  RNRiveErrorType["IncorrectAnimationName"] = "IncorrectAnimationName";
  RNRiveErrorType["MalformedFile"] = "MalformedFile";
  RNRiveErrorType["IncorrectArtboardName"] = "IncorrectArtboardName";
  RNRiveErrorType["IncorrectStateMachineName"] = "IncorrectStateMachineName";
  RNRiveErrorType["IncorrectStateMachineInput"] = "IncorrectStateMachineInput";
})(RNRiveErrorType || (RNRiveErrorType = {}));
