// rive-react-native/lib/commonjs/types.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.RNRiveErrorType = exports.Direction = exports.LoopMode = exports.Alignment = exports.Fit = void 0;
let Fit;
exports.Fit = Fit;

(function (Fit) {
  Fit["Cover"] = "cover";
  Fit["Contain"] = "contain";
  Fit["Fill"] = "fill";
  Fit["FitWidth"] = "fitWidth";
  Fit["FitHeight"] = "fitHeight";
  Fit["None"] = "none";
  Fit["ScaleDown"] = "scaleDown";
})(Fit || (exports.Fit = Fit = {}));

let Alignment;
exports.Alignment = Alignment;

(function (Alignment) {
  Alignment["TopLeft"] = "topLeft";
  Alignment["TopCenter"] = "topCenter";
  Alignment["TopRight"] = "topRight";
  Alignment["CenterLeft"] = "centerLeft";
  Alignment["Center"] = "center";
  Alignment["CenterRight"] = "centerRight";
  Alignment["BottomLeft"] = "bottomLeft";
  Alignment["BottomCenter"] = "bottomCenter";
  Alignment["BottomRight"] = "bottomRight";
})(Alignment || (exports.Alignment = Alignment = {}));

let LoopMode;
exports.LoopMode = LoopMode;

(function (LoopMode) {
  LoopMode["OneShot"] = "oneShot";
  LoopMode["Loop"] = "loop";
  LoopMode["PingPong"] = "pingPong";
  LoopMode["Auto"] = "auto";
})(LoopMode || (exports.LoopMode = LoopMode = {}));

let Direction;
exports.Direction = Direction;

(function (Direction) {
  Direction["Backwards"] = "backwards";
  Direction["Auto"] = "auto";
  Direction["Forwards"] = "forwards";
})(Direction || (exports.Direction = Direction = {}));

let RNRiveErrorType;
exports.RNRiveErrorType = RNRiveErrorType;

(function (RNRiveErrorType) {
  RNRiveErrorType["FileNotFound"] = "FileNotFound";
  RNRiveErrorType["UnsupportedRuntimeVersion"] = "UnsupportedRuntimeVersion";
  RNRiveErrorType["IncorrectRiveFileUrl"] = "IncorrectRiveFileUrl";
  RNRiveErrorType["IncorrectAnimationName"] = "IncorrectAnimationName";
  RNRiveErrorType["MalformedFile"] = "MalformedFile";
  RNRiveErrorType["IncorrectArtboardName"] = "IncorrectArtboardName";
  RNRiveErrorType["IncorrectStateMachineName"] = "IncorrectStateMachineName";
  RNRiveErrorType["IncorrectStateMachineInput"] = "IncorrectStateMachineInput";
})(RNRiveErrorType || (exports.RNRiveErrorType = RNRiveErrorType = {}));
